src/controller/tests/TestAutoCommissioner.cpp

```diff
@@ -338,4 +338,194 @@ TEST_F(AutoCommissionerTest, NextStageConfigureTCAcknowledgments)
     EXPECT_EQ(nextStage, kSendPAICertificateRequest);
 }
 
+TEST_F(AutoCommissionerTest, TrySecondaryNetwork_confirm_true)
+{
+    AutoCommissionerTestAccess privateConfigCommissioner(&mCommissioner);
+    privateConfigCommissioner.TrySecondaryNetwork();
+    EXPECT_EQ(privateConfigCommissioner.TryingSecondaryNetwork(), true);
+}
+
+TEST_F(AutoCommissionerTest, ResetTryingSecondaryNetwork_confirm_false)
+{
+    AutoCommissionerTestAccess privateConfigCommissioner(&mCommissioner);
+    privateConfigCommissioner.ResetTryingSecondaryNetwork();
+    EXPECT_EQ(privateConfigCommissioner.TryingSecondaryNetwork(), false);
+}
+
+TEST_F(AutoCommissionerTest, IsScanNeededCombinations)
+{
+    struct Case
+    {
+        const char * name;
+        bool attemptWiFi;
+        bool attemptThread;
+        EndpointId wifiEndpoint;
+        EndpointId threadEndpoint;
+        bool scanExpected;
+    };
+
+    const Case cases[] = {
+        {
+            .name           = "WiFiAndThreadSet",
+            .attemptWiFi    = true,
+            .attemptThread  = true,
+            .wifiEndpoint   = kRootEndpointId,
```

Comment:
it seems to me that wheever `attemptX` is true we have the `XEndpoint` be kRootEndpointId and otherwise it is kInvalidEndpointId.

Can we remove some of the redundancy in the code?